### PR TITLE
Add EigenValue/Vector for general matrix.

### PR DESCRIPTION
**Pull request by doraneko94**
I added `eig` for calculating eigenvalues/vectors of general matrices using `*geev` of lapack.
Please confirm the performance with `tests/eig.rs` and `examples/eig.rs`.

### CODE REVIEW

File: src/eig.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+//! Eigenvalue decomposition for Hermite matrices
```
Comment (termoshtt) [inline src/eig.rs]:
```suggestion
//! Eigenvalue decomposition for non-symmetric square matrices
```

File: src/lapack/eig.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                        conj = true;
+                        for j in 0..n {
+                            v[n*j+i] = Self::Complex::new(vr[n*j+i], vr[n*j+i+1]);
+                        }
+                    } else {
+                        for j in 0..n {
+                            v[n*j+i] = Self::Complex::new(vr[n*j+i], 0.0);
+                        }
+                    }
+                }
```
Comment (termoshtt) [inline src/lapack/eig.rs]: - Please add a comment about what done here
  - Explanation of `VL` in [LAPACK document](http://www.netlib.org/lapack/explore-html/d9/d8e/group__double_g_eeigen_ga66e19253344358f5dee1e60502b9e96f.html) may be good template.
- Can this loop be rewrite into `for j in 0..n { for i in 0..n { ... } }` (swtich i and j order) ? In general, `v[n*j+i]` access should be continuous.

File: src/eig.rs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+use ndarray::*;
+use crate::error::*;
+use crate::layout::*;
+use crate::types::*;
+
+/// Eigenvalue decomposition of general matrix reference
+pub trait Eig {
+    type EigVal;
+    type EigVec;
+    fn eig(&self) -> Result<(Self::EigVal, Self::EigVec)>;
```
Comment (termoshtt) [inline src/eig.rs]: Please add a comment that this function calculate the **right** eigenvalue

File: src/eig.rs (hunk 1 above)
Comment (doraneko94) [reply src/eig.rs]: Thanks, it was completely typo.

File: src/lapack/eig.rs (hunk 2 above)
Comment (doraneko94) [reply src/lapack/eig.rs]: I rewrote this part using `match`, now we can construct `v: Vec<c64>` continuously.

File: src/eig.rs (hunk 3 above)
Comment (doraneko94) [reply src/eig.rs]: Ok, thanks.

**Comment by doraneko94** [comment]
And I removed an unnecessary semicolon in `examples/solveh.rs`.

**Comment by termoshtt** [comment]
Thanks!

**Pull request merged**